include/xsparse/level_capabilities/co_iteration.hpp

```diff
@@ -0,0 +1,162 @@
+#ifndef XSPARSE_CO_ITERATION_HPP
+#define XSPARSE_CO_ITERATION_HPP
+
+#include <vector>
+#include <tuple>
+
+namespace xsparse::level_capabilities
+{
+    template <class F, class IK, class PK, class Levels, class Is>
+    class Coiterate;
+
+    template <class F, class IK, class PK, class... Levels, class... Is>
+    class Coiterate<F, IK, PK, std::tuple<Levels...>, std::tuple<Is...>>
+    {
+    public:
+        class coiteration_helper
+        {
+        private:
+            std::tuple<Is...> m_i;
```

Comment:
@hameerabbasi If this is inside `iterator`, then do we need to pass these to `iterator`'s constructor as parameters? Because right now we are getting them from `coiteration_helper`'s constructor...

Reply:
That's correct.

Reply:
Actually, no. It should be the `IK`s you get out of the beginning of each iterator.

Reply:
I don't think it needs to be stored either. Unless you mean `IKM1`?